kratos/geometries/geometry.h

```diff
@@ -410,10 +423,93 @@ class Geometry : public PointerVector<TPointType>
     //     return p_clone;
     // }
 
-    //lumping factors for the calculation of the lumped mass matrix
-    virtual Vector& LumpingFactors( Vector& rResult )  const
-    {
-        KRATOS_ERROR << "Called the virtual function for LumpingFactors " << *this << std::endl;
+    /**
+     * @brief Lumping factors for the calculation of the lumped mass matrix
+     * @param rResult Vector containing the lumping factors
+     * @param LumpingMethod The lumping method considered. The three methods available are:
+     *      - The row sum method
+     *      - Diagonal scaling
+     *      - Evaluation of M using a quadrature involving only the nodal points and thus automatically yielding a diagonal matrix for standard element shape function
+     */
+    virtual Vector& LumpingFactors(
+        Vector& rResult,
+        const LumpingMethods LumpingMethod = LumpingMethods::ROW_SUM
+        )  const
+    {
+        const SizeType number_of_nodes = this->size();
+        const SizeType local_space_dimension = this->LocalSpaceDimension();
+
+        // Clear lumping factors
+        if (rResult.size() != number_of_nodes)
+            rResult.resize(number_of_nodes, false);
+        rResult = ZeroVector(number_of_nodes);
+
+        if (LumpingMethod == LumpingMethods::ROW_SUM) {
+            IntegrationMethod integration_method = GetDefaultIntegrationMethod();
+            integration_method = static_cast<IntegrationMethod>(static_cast<int>(integration_method) + 1);
+            const GeometryType::IntegrationPointsArrayType& integration_points = this->IntegrationPoints( integration_method );
+            const Matrix& Ncontainer = this->ShapeFunctionsValues(integration_method);
+
+            for ( IndexType point_number = 0; point_number < integration_points.size(); ++point_number ) {
+                const double detJ = DeterminantOfJacobian(point_number, integration_method);
+                const double integration_weight = integration_points[point_number].Weight() * detJ;
+                const Vector& rN = row(Ncontainer,point_number);
+
+                for ( IndexType i = 0; i < number_of_nodes; ++i ) {
+                    for ( IndexType j = 0; j < number_of_nodes; ++j ) {
+                        const double NiNj_weight = rN[i] * rN[j] * integration_weight;
+                        rResult[i] += NiNj_weight;
+                    }
+                }
+            }
+
+            // Divide by the domain size
+            const double domain_size = DomainSize();
+            for ( IndexType i = 0; i < number_of_nodes; ++i ) {
+                rResult[i] /= domain_size;
```

Comment:
this is needed as it is defined now, however i would be in favor of changing the behaviour here. (to be discussed of course, so the change is not for now, and not in this PR)

Reply:
OK

Reply:
the more i think of it, the more i believe we should change the signature of this function, even as part of this PR.

My proposal would be to change the behaviour so that 
1 - it does not do the final division by domain_size (btw, it is not a good implementation to call DomainSize() here, here, better sum up the weights!!)
2 - i would return the value of domain_size

Reply:
The thing is with the current implementation and to not break any existing code I needed to divide by the DomainSize()

Reply:
doing so is quite inefficient since it implies computing twice the determinants of the jacobian

if you add 

~~~
     double domain_size = 0.0;
      for ( IndexType point_number = 0; point_number < integration_points.size(); ++point_number ) {
            domain_size += integration_points[point_number].Weight() * detJ_vector[point_number];

you can avoid that cost